### PR TITLE
Confirm if user wants example when creating app

### PR DESCRIPTION
**Pull request by merelinguist**
Currently, when a user runs `create-next-app`, it automatically generates a Next project from a blank template. This pull adds an extra question to the creation process, which confirms whether they want to use an example:

carbon

This was [suggested](https://github.com/zeit/next.js/issues/8090#issuecomment-548368124) by @leerob in #8090.

> The first time I ran this I forgot about the examples flag and then had to re-run it because I wanted the styled-components example. I think it would help 😄

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| buildDuration | 11.7s | 11.2s | -479ms |
| nodeModulesSize | 52.7 MB | 52.7 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.13 kB | 5.13 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..54d3.js gzip | 4.68 kB | 4.68 kB | ✓ |
| commons.HASH.js gzip | 4.06 kB | 4.06 kB | ✓ |
| de003c3a9d30..d239.js gzip | 16.2 kB | 16.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 70 kB | 70 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.13 kB | 4.13 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..dule.js gzip | 5.56 kB | 5.56 kB | ✓ |
| de003c3a9d30..dule.js gzip | 15.1 kB | 15.1 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 64.6 kB | 64.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 4.76 kB | 4.76 kB | ✓ |
| Overall change | 4.76 kB | 4.76 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _app.js gzip | 1.15 kB | 1.15 kB | ✓ |
| _error.js gzip | 4.07 kB | 4.07 kB | ✓ |
| hooks.js gzip | 779 B | 779 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.89 kB | 2.89 kB | ✓ |
| routerDirect.js gzip | 283 B | 283 B | ✓ |
| withRouter.js gzip | 282 B | 282 B | ✓ |
| Overall change | 9.68 kB | 9.68 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _app.module.js gzip | 576 B | 576 B | ✓ |
| _error.module.js gzip | 3.06 kB | 3.06 kB | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 2.46 kB | 2.46 kB | ✓ |
| routerDirect..dule.js gzip | 273 B | 273 B | ✓ |
| withRouter.m..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 7.22 kB | 7.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| index.html gzip | 1.03 kB | 1.03 kB | ✓ |
| link.html gzip | 1.04 kB | 1.04 kB | ✓ |
| withRouter.html gzip | 1.03 kB | 1.03 kB | ✓ |
| Overall change | 3.11 kB | 3.11 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| buildDuration | 12.2s | 13s | ⚠️ +815ms |
| nodeModulesSize | 52.7 MB | 52.7 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.13 kB | 5.13 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..54d3.js gzip | 4.68 kB | 4.68 kB | ✓ |
| commons.HASH.js gzip | 4.06 kB | 4.06 kB | ✓ |
| de003c3a9d30..d239.js gzip | 16.2 kB | 16.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 70 kB | 70 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.13 kB | 4.13 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..dule.js gzip | 5.56 kB | 5.56 kB | ✓ |
| de003c3a9d30..dule.js gzip | 15.1 kB | 15.1 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 64.6 kB | 64.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 4.76 kB | 4.76 kB | ✓ |
| Overall change | 4.76 kB | 4.76 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _app.js gzip | 1.15 kB | 1.15 kB | ✓ |
| _error.js gzip | 4.07 kB | 4.07 kB | ✓ |
| hooks.js gzip | 779 B | 779 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.89 kB | 2.89 kB | ✓ |
| routerDirect.js gzip | 283 B | 283 B | ✓ |
| withRouter.js gzip | 282 B | 282 B | ✓ |
| Overall change | 9.68 kB | 9.68 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _app.module.js gzip | 576 B | 576 B | ✓ |
| _error.module.js gzip | 3.06 kB | 3.06 kB | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 2.46 kB | 2.46 kB | ✓ |
| routerDirect..dule.js gzip | 273 B | 273 B | ✓ |
| withRouter.m..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 7.22 kB | 7.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _error.js gzip | 290 kB | 290 kB | -269 B |
| 404.html gzip | 1.44 kB | 1.44 kB | ✓ |
| hooks.html gzip | 1.08 kB | 1.08 kB | ✓ |
| index.js gzip | 289 kB | 289 kB | -58 B |
| link.js gzip | 319 kB | 319 kB | -229 B |
| routerDirect.js gzip | 316 kB | 316 kB | ⚠️ +386 B |
| withRouter.js gzip | 315 kB | 316 kB | ⚠️ +859 B |
| Overall change | 1.53 MB | 1.53 MB | ⚠️ +689 B |

</details>

</details>


Commit: 8f192de436fdf5107ef6ec8affdd2c6eb49ea5f8

**Review by lfades — CHANGES_REQUESTED** [review]
Some things I don't like about this change:
- Even though this may be helpful, it introduces a new question, meaning more time required to start the app
- We have over 200 examples, and every example includes a download link with `create-next-app`, afaik when we add a reference to `create-next-app` it either already includes the name of the example or it doesn't because we want the user to download the default example (and that example is a good introduction to Next.js).

Alternative: Ask for the name of the example if the user wants an example, but doesn't know which one, e.g: `yarn next-app --example`

**Comment by merelinguist** [comment]
Screenshot 2020-02-17 at 20 15 03

@lfades Just FYI this is what the Gatsby CLI does, but I 100% see where you’re coming from as far as simplicity/speed goes :).  I’ve amended as instructed, let me know what you think.

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| buildDuration | 12s | 12.1s | ⚠️ +145ms |
| nodeModulesSize | 52.7 MB | 52.7 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.14 kB | 5.14 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..54d3.js gzip | 4.68 kB | 4.68 kB | ✓ |
| commons.HASH.js gzip | 4.06 kB | 4.06 kB | ✓ |
| de003c3a9d30..4e43.js gzip | 16.2 kB | 16.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 70 kB | 70 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.14 kB | 4.14 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..dule.js gzip | 5.56 kB | 5.56 kB | ✓ |
| de003c3a9d30..dule.js gzip | 15.1 kB | 15.1 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 64.7 kB | 64.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 4.76 kB | 4.76 kB | ✓ |
| Overall change | 4.76 kB | 4.76 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _app.js gzip | 1.15 kB | 1.15 kB | ✓ |
| _error.js gzip | 4.07 kB | 4.07 kB | ✓ |
| hooks.js gzip | 779 B | 779 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.89 kB | 2.89 kB | ✓ |
| routerDirect.js gzip | 283 B | 283 B | ✓ |
| withRouter.js gzip | 282 B | 282 B | ✓ |
| Overall change | 9.68 kB | 9.68 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _app.module.js gzip | 576 B | 576 B | ✓ |
| _error.module.js gzip | 3.06 kB | 3.06 kB | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 2.46 kB | 2.46 kB | ✓ |
| routerDirect..dule.js gzip | 273 B | 273 B | ✓ |
| withRouter.m..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 7.22 kB | 7.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| index.html gzip | 1.03 kB | 1.03 kB | ✓ |
| link.html gzip | 1.04 kB | 1.04 kB | ✓ |
| withRouter.html gzip | 1.03 kB | 1.03 kB | ✓ |
| Overall change | 3.11 kB | 3.11 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| buildDuration | 13.3s | 13.2s | -36ms |
| nodeModulesSize | 52.7 MB | 52.7 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.14 kB | 5.14 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..54d3.js gzip | 4.68 kB | 4.68 kB | ✓ |
| commons.HASH.js gzip | 4.06 kB | 4.06 kB | ✓ |
| de003c3a9d30..4e43.js gzip | 16.2 kB | 16.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 70 kB | 70 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.14 kB | 4.14 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..dule.js gzip | 5.56 kB | 5.56 kB | ✓ |
| de003c3a9d30..dule.js gzip | 15.1 kB | 15.1 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 64.7 kB | 64.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 4.76 kB | 4.76 kB | ✓ |
| Overall change | 4.76 kB | 4.76 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _app.js gzip | 1.15 kB | 1.15 kB | ✓ |
| _error.js gzip | 4.07 kB | 4.07 kB | ✓ |
| hooks.js gzip | 779 B | 779 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.89 kB | 2.89 kB | ✓ |
| routerDirect.js gzip | 283 B | 283 B | ✓ |
| withRouter.js gzip | 282 B | 282 B | ✓ |
| Overall change | 9.68 kB | 9.68 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _app.module.js gzip | 576 B | 576 B | ✓ |
| _error.module.js gzip | 3.06 kB | 3.06 kB | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 2.46 kB | 2.46 kB | ✓ |
| routerDirect..dule.js gzip | 273 B | 273 B | ✓ |
| withRouter.m..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 7.22 kB | 7.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall decrease ✓</summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _error.js gzip | 289 kB | 290 kB | ⚠️ +593 B |
| 404.html gzip | 1.45 kB | 1.45 kB | ✓ |
| hooks.html gzip | 1.08 kB | 1.08 kB | ✓ |
| index.js gzip | 289 kB | 289 kB | -187 B |
| link.js gzip | 319 kB | 319 kB | -638 B |
| routerDirect.js gzip | 316 kB | 315 kB | -1.05 kB |
| withRouter.js gzip | 317 kB | 316 kB | -393 B |
| Overall change | 1.53 MB | 1.53 MB | -1.67 kB |

</details>

</details>


Commit: 5952dfa311296c8c1271004dc28c64c09fa3af12

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| buildDuration | 9.8s | 10.3s | ⚠️ +439ms |
| nodeModulesSize | 53.8 MB | 53.8 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.05 kB | 5.05 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..abdd.js gzip | 12.2 kB | 12.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.13 kB | 4.13 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 9.2 kB | 9.2 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 53.2 kB | 53.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 20.3 kB | 20.3 kB | ✓ |
| Overall change | 20.3 kB | 20.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _app.js gzip | 1.06 kB | 1.06 kB | ✓ |
| _error.js gzip | 2.98 kB | 2.98 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 1.89 kB | 1.89 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.37 kB | 7.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _app.module.js gzip | 577 B | 577 B | ✓ |
| _error.module.js gzip | 2.08 kB | 2.08 kB | ✓ |
| hooks.module.js gzip | 370 B | 370 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 1.47 kB | 1.47 kB | ✓ |
| routerDirect..dule.js gzip | 271 B | 271 B | ✓ |
| withRouter.m..dule.js gzip | 270 B | 270 B | ✓ |
| Overall change | 5.25 kB | 5.25 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| index.html gzip | 926 B | 926 B | ✓ |
| link.html gzip | 933 B | 933 B | ✓ |
| withRouter.html gzip | 922 B | 922 B | ✓ |
| Overall change | 2.78 kB | 2.78 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| buildDuration | 10.8s | 10.5s | -266ms |
| nodeModulesSize | 53.8 MB | 53.8 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.05 kB | 5.05 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..abdd.js gzip | 12.2 kB | 12.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57.2 kB | 57.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.13 kB | 4.13 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 9.2 kB | 9.2 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 53.2 kB | 53.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 20.3 kB | 20.3 kB | ✓ |
| Overall change | 20.3 kB | 20.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _app.js gzip | 1.06 kB | 1.06 kB | ✓ |
| _error.js gzip | 2.98 kB | 2.98 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 1.89 kB | 1.89 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.37 kB | 7.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _app.module.js gzip | 577 B | 577 B | ✓ |
| _error.module.js gzip | 2.08 kB | 2.08 kB | ✓ |
| hooks.module.js gzip | 370 B | 370 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 1.47 kB | 1.47 kB | ✓ |
| routerDirect..dule.js gzip | 271 B | 271 B | ✓ |
| withRouter.m..dule.js gzip | 270 B | 270 B | ✓ |
| Overall change | 5.25 kB | 5.25 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall decrease ✓</summary>

|  | zeit/next.js canary  | merelinguist/next.js cli-create-from-example | Change |
| - | - | - | - |
| _error.js gzip | 289 kB | 289 kB | -251 B |
| 404.html gzip | 1.33 kB | 1.33 kB | ✓ |
| hooks.html gzip | 965 B | 965 B | ✓ |
| index.js gzip | 289 kB | 290 kB | ⚠️ +248 B |
| link.js gzip | 299 kB | 297 kB | -1.37 kB |
| routerDirect.js gzip | 297 kB | 296 kB | -597 B |
| withRouter.js gzip | 296 kB | 296 kB | ⚠️ +310 B |
| Overall change | 1.47 MB | 1.47 MB | -1.66 kB |

</details>

</details>


Commit: 83d19e992f0db3e936d6f48d81c721916a594ce8

**Review by lfades — APPROVED** [review]
LGTM, I've updated the PR so I'll wait for the review of someone else from the team

**Comment by timneutkens** [comment]
If it's asked under `create-next-app --example` it's unlikely anyone will use it as it's not easy to know it exists.

**Comment by lfades** [comment]
@timneutkens  "I'm not opposed to asking it as part of the normal flow": Should we keep it as an additional step as introduced initially in this PR?

**Comment by timneutkens** [comment]
> @timneutkens "I'm not opposed to asking it as part of the normal flow": Should we keep it as an additional step as introduced initially in this PR?

Yeah there's basically 2 options:
- Add it as part of the default flow unless `--example <something>` is provided explicitly
- Don't add this feature

Hiding it under `--example` will cause it to not be used by users.

**Comment by lfades** [comment]
@timneutkens I've updated the PR to ask for a template, it can be the default starter app or an example.

**Comment by timneutkens** [comment]
@lfades could you add a gif/video of what it will look like

**Comment by merelinguist** [comment]
@timneutkens 

See these two demos:

- https://terminalizer.com/view/cfe3d5463600

- https://terminalizer.com/view/a47a763d3601

**Comment by timneutkens** [comment]
Nice! Looks great!

**Comment by lfades** [comment]
Did some UX changes, the PR is now ready for merge

File: test/integration/create-next-app/index.test.js [hunk 1]
```diff
@@ -150,4 +162,43 @@ describe('create next app', () => {
...
     ).toBeTruthy()
   })
+
+  it('should allow to manually select an example', async () => {
+    const runExample = (...args) => {
+      const res = run(...args)
+
+      function pickExample(data) {
+        if (/hello-world/.test(data.toString())) {
+          console.log('YO')
```
Comment (Timer) [inline test/integration/create-next-app/index.test.js:172]: Left over debugging

File: packages/create-next-app/helpers/examples.ts [hunk 2]
```diff
@@ -69,3 +69,10 @@ export function downloadAndExtractExample(
     tar.extract({ cwd: root, strip: 3 }, [`next.js-canary/examples/${name}`])
   )
 }
+
+export async function listExamples(): Promise<any> {
+  const res = await got(
+    'https://api.github.com/repositories/70107786/contents/examples'
+  ).catch(e => e)
```
Comment (Timer) [inline packages/create-next-app/helpers/examples.ts:76]: When this fails we need to skip prompting the user to select an example. Currently, they'll get junk output.

**Review by Timer — CHANGES_REQUESTED** [review]

**Review by Timer — APPROVED** [review]

**Pull request merged**